main.py

```diff
@@ -35,66 +35,85 @@ def default_config():
             }
 
 
-def default_experiment(args):
-    if args.experiment == 'custom':
-        return args
-
-    # TO FILL IN FOR ALL THE DEFAULT HYPERPARAMETERS OF THE EXPERIMENTS WE WANT
-    # TO REPLICATE
-    # @aleco
-
-    return args
+def set_experiment(default_config):
+    """ Set up default experiments to replicate the results in the paper:
+        "Understanding Disentanglement in Beta-VAE" (https://arxiv.org/pdf/1804.03599.pdf)
+    """
+    if default_config['experiment'] == 'custom':
+        return default_config
+    elif default_config['experiment'] == 'vae_blob_x_y':
+        default_config['capacity'] = 1
+        default_config['dataset'] = 'dsprites'
+    elif default_config['experiment'] == 'beta_vae_blob_x_y':
+        default_config['capacity'] = 150
+        default_config['dataset'] = 'black_and_white_dsprite'
+    elif default_config['experiment'] == 'beta_vae_dsprite':
+        default_config['capacity'] = [0.0, 25.0, 100000, 1000.0]
+        default_config['dataset'] = 'dsprites'
+    elif default_config['experiment'] == 'beta_vae_celeba':
+        default_config['capacity'] = [0.0, 50.0, 100000, 1000]
+        default_config['dataset'] = 'celeba'
+    elif default_config['experiment'] == 'beta_vae_colour_dsprite':
+        default_config['capacity'] = [0.0, 25.0, 100000, 1000]
+        default_config['dataset'] = 'dsprites'
+    elif default_config['experiment'] == 'beta_vae_chairs':
+        default_config['capacity'] = 1000
+        default_config['dataset'] = 'chairs'
+
+    return default_config
 
 
 def parse_arguments():
-    defaultsConfig = default_config()
+    default_config = default_experiment()
+    experiment_config = set_experiment(default_config)
```

Comment:
have you actually tested this ? by looking quickly at it, it seems that you are not doing setting the experiments. I.e I believe you have to use the default_config as I was doing and at the end of the function use `args = default_experiment(args)` (sorry I thought I added it). Because right now you are setting the experiment config before setting the experiments (or maybe I overlooked something), meaning that the experiments config that will be used are always the same corresponding to the default experiment.

Reply:
I have tested and corrected this now. Thanks!